File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.swift
class Solution {
    func minSubArrayLen(_ t: Int, _ n: [Int]) -> Int {
        var size = Int.max, start = 0, sum = 0
        for (a,b) in n.enumerated() {
            sum += b
            while sum >= t, start <= a {
                size = min(size, a - start + 1)
                sum -= n[start]
                start += 1
            }
        }
        return size == .max ? 0 : size
    }
}

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
#include<vector>
#include<climits>
using namespace std;

class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int left = 0, right = 0;
        int sum = 0;
        int min_len = INT_MAX;
        
        while(right < nums.size()){
            sum += nums[right];
            
            while(sum >= target){
                min_len = min(min_len, right - left + 1);
                sum -= nums[left];
                left++;
            }
            right++;
        }
        return (min_len != INT_MAX) ? min_len : 0;
    }
};
